binder/src/main/java/io/grpc/binder/BinderChannelBuilder.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.binder;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import android.app.Application;
+import android.content.ComponentName;
+import android.content.Context;
+import androidx.core.content.ContextCompat;
+import io.grpc.ChannelCredentials;
+import io.grpc.ChannelLogger;
+import io.grpc.CompressorRegistry;
+import io.grpc.DecompressorRegistry;
+import io.grpc.ForwardingChannelBuilder;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.binder.internal.BinderTransport;
+import io.grpc.internal.ClientTransportFactory;
+import io.grpc.internal.ConnectionClientTransport;
+import io.grpc.internal.FixedObjectPool;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.ManagedChannelImplBuilder;
+import io.grpc.internal.ManagedChannelImplBuilder.ClientTransportFactoryBuilder;
+import io.grpc.internal.ObjectPool;
+import io.grpc.internal.SharedResourcePool;
+import java.net.SocketAddress;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Builder for a gRPC channel which communicates with an Android bound service.
+ *
+ * @see <a href="https://developer.android.com/guide/components/bound-services.html">Bound
+ *     Services</a>
+ */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/8022")
+public final class BinderChannelBuilder
+    extends ForwardingChannelBuilder<BinderChannelBuilder> {
+
+  /**
+   * Creates a channel builder that will bind to a remote Android service.
+   *
+   * <p>The underlying Android binding will be torn down when the channel becomes idle. This happens
+   * after 30 minutes without use by default but can be configured via {@link
+   * ManagedChannelBuilder#idleTimeout(long, TimeUnit)} or triggered manually with {@link
+   * ManagedChannel#enterIdle()}.
+   *
+   * <p>You the caller are responsible for managing the lifecycle of any channels built by the
+   * resulting builder. They will not be shut down automatically.
+   *
+   * @param sourceContext the context to bind from (e.g. The current Activity or Application).
+   * @param targetAddress the {@link AndroidComponentAddress} referencing the service to bind to.
+   * @return a new builder
+   */
+  public static BinderChannelBuilder create(
```

Comment:
We need to add `forAddress()` and `forTarget()` static methods here to hide/shadow the ones in ManagedChannelBuilder, but we just have them throw. See in-process transport for an example. We'll need to do the equivalent on server-side as well.

Reply:
Done.